### PR TITLE
Allow multipart/form-data in PATCH request

### PR DESCRIPTION
**Pull request by sabandi**

### CODE REVIEW

**Comment by sabandi** [comment]
@jeevatkm could you please review this change.

**Comment by moorereason** [comment]
Seems simple enough.  We obviously don't have tests around this code.  Do you have time to add tests?

**Comment by sabandi** [comment]
> Seems simple enough. We obviously don't have tests around this code. Do you have time to add tests?

@moorereason Added a simple test as per your suggestion. Please review.

**Comment by sabandi** [comment]
Test result.
```
~/Desktop/ patch-1 ❯ go test  -run ^TestMultiPartUploadFileViaPatch$ ./...
ok      github.com/go-resty/resty/v2    0.202s
```

**Comment by sabandi** [comment]
@moorereason just checking in to see if you got a chance to review this.

File: middleware.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func parseRequestHeader(c *Client, r *Request) error {
 func parseRequestBody(c *Client, r *Request) (err error) {
 	if isPayloadSupported(r.Method, c.AllowGetMethodPayload) {
 		// Handling Multipart
-		if r.isMultiPart && !(r.Method == MethodPatch) {
```
Comment (jeevatkm) [inline middleware.go:131]: @sabandi HTTP spec is keep on getting better!

**Review by jeevatkm — APPROVED** [review]
@sabandi Thanks you for your PR and contribution. I'm sorry for the delayed attention on the PR.

**Pull request merged**